docs/photutils/psf.rst

```diff
@@ -28,10 +28,379 @@ model-fitting analysis, regardless to exactly what kind of model is actually
+seminal paper <http://adsabs.harvard.edu/abs/1987PASP...99..191S >`_ for 
+crowded-field stellar photometry.
+The DAOPHOT algorithm consists in applying the loop FIND, GROUP, NSTAR,
+SUBTRACT, FIND until no more stars are detected or a given number of
+iterations is reached.
+
+Basically, DAOPHOT works as follows. The first step is to estimate the sky
+background. For this task, photutils provides several classes to compute
+scalar and 2D backgrounds, see `~photutils.background` for details. The next
+step is to find an initial estimate of the positions of potential sources.
+This can be accomplished by using source detection algorithms,
+which are implemented in `~photutils.detection`.
+
+After finding sources one would apply a clustering algorithm in order to label
+the sources according to groups. Usually, those groups are formed by a
+distance criterion, which is the case of the grouping algorithm proposed
+by Stetson. In `~photutils.psf.DAOGroup`, we provide an implementation of
+that algorithm.
+
+Provided that the groups are available, the next step is to fit the sources
+simultaneously for each group. This task can be done using an astropy
+fitter, for instance, `~astropy.modeling.fitting.LevMarLSQFitter`.
+
+After sources are fitted, they are subtracted from the given image
+and, after fitting all sources, the residual image is analyzed by the finding
+routine again in order to check if there exist any source which has not been
+detected previously. This process goes on until no more sources are identified 
+by the finding routine.
+
+.. note::
+    It is important to note the conventions on the column names of the
+    input/output astropy Tables which are passed along the source detection
+    objects and the photometry objetcs. For instance, all source detection
+    objects should output a table with columns named as ``xcentroid`` and
+    ``ycentroid`` (check `~photutils.detection`). On the other hand,
+    `~photutils.psf.DAOGroup` expects columns named as ``x_0`` and ``y_0``,
+    which represents the initial guesses on the sources' centroids.
+    Finally, the output of the fitting process shows columns named as
+    ``x_fit``, ``y_fit``, ``flux_fit`` for the optimum values and
+    ``x_0``, ``y_0``, ``flux_0`` for the initial guesses.
+    Although this convention implies that the columns have to be renamed
+    along the process, it has the advantage of clarity so that one can
+    keep track and easily differentiate input/outputs.
+
+
+Basic Usage
+^^^^^^^^^^^
+
+The `~photutils.psf.DAOPhotPSFPhotometry` is the core class to implement the
+DAOPHOT algorithm to perform PSF photometry in crowded fields.
+
+It basically encapsulates the loop "FIND, GROUP, NSTAR, SUBTRACT" in one place
+so that one can easily perform PSF photometry just by setting up a
+`~photutils.psf.DAOPhotPSFPhotometry` object.
+
+This class was implemented in such a way that it can be used as a callable
+function. The basic idea is illustrated as follows:
```

Comment:
Maybe also just mention briefly here that the "guts" of it are in `do_photometry`.  This is a useful thing because otherwise novices might not understand why e.g. the docstring of `__call__` doesn't have any of the details of how to use it.